src/coreclr/src/jit/flowgraph.cpp

```diff
@@ -23820,7 +23848,8 @@ Statement* Compiler::fgInlinePrependStatements(InlineInfo* inlineInfo)
 
                 GenTree* argSingleUseNode = argInfo.argBashTmpNode;
 
-                if ((argSingleUseNode != nullptr) && !(argSingleUseNode->gtFlags & GTF_VAR_CLONED) && argIsSingleDef)
+                if ((argSingleUseNode != nullptr) && !(argSingleUseNode->gtFlags & GTF_VAR_CLONED) && argIsSingleDef &&
+                    !argHasPutArg)
```

Comment:
This new condition allows us to always the argument as a temp with the correct type, we need it because for a correct direct substitution we need to know if the parent is a call or not but we don't have this information.
We create the node when we see an IL opcode like `ldlocal` and don't have parent information, later, when the node is used, we can't distinguish it from a regular `LCL_VAR`.
The change causes a small regression on all platforms:
```
windows x64 crossgen: 5134 (0.02% of base)
windows x64 pmi:  14911 (0.03% of base)
windows arm64 crossgen: 5688 (0.01% of base)
linux x64 crossgen: 9036 (0.02% of base)

```
, etc. all lower than 0.04%.

What happens there is:
1. during importation we create a `LCL_VAR tempForOurArg`, save a pointer to it;
2. in `fgInsertInlineeBlocks` we iterate over all such pointers and do in-place replacement if there was only one use of the lclVar, but we don't know the user, so we can't say if it is a call that need a `PUTARG` or not.

I have not found a simple way to avoid it or compensate for it.